File: common/src/main/java/net/impactdev/gts/common/blacklist/BlacklistImpl.java
package net.impactdev.gts.common.blacklist;

import com.google.common.collect.ArrayListMultimap;
import com.google.common.collect.Multimap;
import net.impactdev.gts.api.blacklist.Blacklist;
import net.kyori.adventure.key.Key;

public class BlacklistImpl implements Blacklist {

    private final Multimap<Class<?>, Key> blacklist = ArrayListMultimap.create();

    @Override
    public Multimap<Class<?>, Key> getBlacklist() {
        return this.blacklist;
    }

    @Override
    public void append(Class<?> registrar, Key key) {
        this.blacklist.put(registrar, key);
    }

    @Override
    public boolean isBlacklisted(Class<?> registrar, Key query) {
        return this.blacklist.get(registrar).contains(query);
    }

    @Override
    public void clear() {
        this.blacklist.clear();
    }

}

File: common/src/main/kotlin/net/impactdev/gts/common/blacklist/BlacklistImpl.kt
package net.impactdev.gts.common.blacklist

import com.google.common.collect.ArrayListMultimap
import com.google.common.collect.Multimap
import net.impactdev.gts.api.blacklist.Blacklist

class BlacklistImpl : Blacklist {
    private val blacklist: Multimap<Class<*>, String> = ArrayListMultimap.create()
    override fun getBlacklist(): Multimap<Class<*>, String> {
        return blacklist
    }

    override fun append(registrar: Class<*>, key: String) {
        blacklist.put(registrar, key)
    }

    override fun isBlacklisted(registrar: Class<*>, query: String): Boolean {
        return blacklist[registrar].contains(query)
    }

    override fun clear() {
        blacklist.clear()
    }
}